src/services/auth.services.ts

```diff
@@ -69,4 +78,51 @@ const registerService = async ({
     return null;
   }
 };
-export { loginService, registerService };
+
+const googleLoginService = async (id_token: string): Promise<AuthReturnType> => {
+
+  const credential = await googleVerify(id_token) as unknown as GooglePayload
+
+  if (!credential) throw new Error("Google verification has failed!");
+
+  const {
+    email,
+    given_name,
+  } = credential;
+
+  try {
+    const user = await UserModel.findOne({ email: email });
+
+    if (!user) {
+      const data = {
+        name: given_name,
+        username: given_name,
+        email: email,
+        password: ":P",
+      };
+
+      const userNew = new UserModel(data);
+
+      const userNewFinish = await userNew.save()
+
+      const token = await generateToken(userNewFinish.email);
```

Comment:
i guess for generatetoken function you can take out await keyword because it's useless there because generateToken is not returning any promise